app/src/main/java/dev/hadrosaur/videodecodeencodedemo/VideoHelpers/SpeedyMediaClock.kt

```diff
@@ -26,7 +26,7 @@ import com.google.android.exoplayer2.util.MediaClock
 class SpeedyMediaClock: MediaClock {
     val MIN_FPS: Int = 1
     private var internalPlaybackParameters = PlaybackParameters(1f)
-    var lastProcessedFrameUs = 0L
+    var lastProcessedFrameUs = -100000L
```

Comment:
Is there a specific reason for -100000? I wonder if we should use Long.MIN_VALUE here?

Can we also add a comment to say why? "Long.MIN_VALUE // initialize to a large negative number to ensure...

Reply:
I used -100000 specifically here, as that is the step multiple value we are using to process the next frame (refer line #39)

We can use anything here just -1, or Long.Min_Value
Just need to make sure that lastProcessedFrameUs needs to evaluate to 0 post processing the first frame. 
So using -1 or Long.Min_value would require a simple if check in getPositionUs for first frame.

Reply:
I see, so if I am understanding correctly, the bug is that there is a logical error so that if `getPositionUs` is called before the first frame has been processed, it is currently returning 100,000, instead of 0?

I think your patch fixes the problem but I see the potential for future bugs here so I'd like to fix the underlying wonky logic and make this clearer/less error prone.

Looking at this, I'm not not convinced about the original `getPositionUs` implementation itself (I think if MIN_FPS is anything other than 1 we will get a int floor which is not going to work). Let me take a different stab at this.

Reply:
Yeah that sounds right.